### PR TITLE
Remove unnecessary disabling of linting errors

### PR DESCRIPTION
**Pull request by Yuyuutsu**
### What is the context of this PR?
This PR removes the unnecessary disabling of linting errors for example `# pylint: disable=protected-access`.
Updated MyPy and Pylint config files to detect unnecessary disabling

Trello Card is [here](https://trello.com/c/lWa9P4j1/5938-remove-unnecessary-disabling-of-linting-errors-in-runner-1-sp)
### How to review
Run `Make lint-python` or `Make lint` and check to see if there are any linting errors
Check if all comments relating to the removed disabling errors are removed

### Checklist
* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Comment by Yuyuutsu** [comment]
[Remove unnecessary disabling of linting errors in Runner (1 SP)](https://trello.com/c/lWa9P4j1/5938-remove-unnecessary-disabling-of-linting-errors-in-runner-1-sp)

File: app/utilities/metadata_parser_v2.py [hunk 1]
```diff
@@ -70,32 +70,25 @@ def validate_receipting_keys(self, data, **kwargs):
...
-    account_service_url = VALIDATORS["url"](required=True)  # type:ignore
-
-    language_code = VALIDATORS["string"](required=False)  # type:ignore
-    channel = VALIDATORS["string"](
-        required=False, validate=validate.Length(min=1)
-    )  # type:ignore
+    )
+    schema_name = VALIDATORS["string"](required=False)
+    schema_url = VALIDATORS["url"](required=False)
+    response_id = VALIDATORS["string"](required=True)
```
Comment (katie-gardner) [inline app/utilities/metadata_parser_v2.py:82]: looks like `cir_instrument_id` got deleted here

Comment (Yuyuutsu) [reply app/utilities/metadata_parser_v2.py:83]: Thanks, I was wondering what was missing. Reverted 👍

File: app/views/handlers/individual_response.py [hunk 2]
```diff
@@ -175,7 +175,7 @@ def rendered_block(self) -> dict:
     def placeholder_renderer(self) -> PlaceholderRenderer:
         return PlaceholderRenderer(
             # Type ignore: Language is defaulted via handle_language in the individual_response blueprint before_request which triggers this
```
Comment (katie-gardner) [inline app/views/handlers/individual_response.py:177]: can drop this type ignore

File: app/forms/questionnaire_form.py [hunk 3]
```diff
@@ -411,7 +411,6 @@ def _get_calculation_total(
         return result
 
     def answers_all_valid(self, answer_id_list: Sequence[str]) -> bool:
-        # pylint: disable=no-member
         # wtforms Form parents are not discoverable in the 2.3.3 implementation
```
Comment (katie-gardner) [inline app/forms/questionnaire_form.py:414]: theres inconsistency with whether the wtf forms being discoverable comment is removed or not, can delete the others too

**Review by katie-gardner — APPROVED** [review]
Spotted two tiny non blocking things and otherwise looking good 👍

File: app/views/handlers/individual_response.py (hunk 2 above)
Comment (Yuyuutsu) [reply app/views/handlers/individual_response.py]: Removed

File: app/forms/questionnaire_form.py (hunk 3 above)
Comment (Yuyuutsu) [reply app/forms/questionnaire_form.py]: Removed all wtf form comments relating to this 👍

File: app/forms/fields/month_year_date_field.py [hunk 4]
```diff
@@ -18,7 +18,6 @@ class YearMonthDateForm(Form):
 
         @cached_property
         def data(self) -> str | None:
-            # pylint: disable=no-member
             # wtforms Form parents are not discoverable in the 2.3.3 implementation
```
Comment (berroar) [inline app/forms/fields/month_year_date_field.py:21]: Can these wtf comments be removed too?

Comment (Yuyuutsu) [reply app/forms/fields/month_year_date_field.py:21]: Removed 👍

**Comment by Yuyuutsu** [comment]
> > When I run make lint I get a `useless-suppression` warning for `too-many-function-args` in `test_schema.py`, however the linter still passes - so maybe why this was missed?
> 
> I also get this error too: Screenshot 2023-11-29 at 17 31 55

Looks like it got reverted when I was fixing a merging issue. Should be good now 👍

**Review by VirajP1002 — COMMENTED** [review]
Both `make lint-python` and `make lint` work for me as intended 👍

**Review by rmccar — APPROVED** [review]

**Pull request merged**